library/aesni.h

```diff
@@ -35,13 +35,13 @@
 /* Can we do AESNI with inline assembly?
  * (Only implemented with gas syntax, only for 64-bit.)
  */
-#if defined(MBEDTLS_HAVE_ASM) && defined(__GNUC__) && \
-    (defined(__amd64__) || defined(__x86_64__))   &&  \
-    !defined(MBEDTLS_HAVE_X86_64)
+#if !defined(MBEDTLS_HAVE_X86_64) && \
+    (defined(__amd64__) || defined(__x86_64__) || \
+    defined(_M_X64) || defined(_M_AMD64))
 #define MBEDTLS_HAVE_X86_64
 #endif
 
-#if defined(MBEDTLS_AESNI_C)
+#if defined(MBEDTLS_AESNI_C) && defined(MBEDTLS_HAVE_X86_64)
```

Comment:
The check of `MBEDTLS_HAVE_X86_64` would prevent 32-bit builds from using AES-NI, which should be a feature in `mbedtls-3.4.0`, see https://github.com/Mbed-TLS/mbedtls/blob/5d396327a71e0e9b02cefd40cf2930c661e62988/ChangeLog#L94-L98
Is this expected?